build.js

```diff
@@ -120,32 +122,35 @@ await remove("./dist", {
 	force: true,
 });
 
+await copy("./src", "./dist", {
+	recursive: true,
+	force: true
+});
+
+if ((await getAllFilesInDirectoryWithExtension("./dist", ".ts")).length !== 0) {
+	try {
+		execSync('tsc --build');
+	}
+	catch (e) {
+		console.error('TypeScript compilation failed with the following error:');
+		console.log(e.stdout.toString()); // Print TypeScript error output
+		console.log(e.stderr.toString()); // Print additional error details if available
+	}
+}
 
 if (!BUNDLE_FILES) {
 	/**
 	 * In development, copy all clientside files directly over to /dist AS THEY ARE.
 	 * No files are bundled. This means a LOT of requests! But, all our comments are there!
```

Comment:
This is the comment that needs to be moved above and modified to match the context.

Also this block is completely empty and no longer needed.